CloudNotes/CloudNotes/Model/NoteManager.swift

```diff
@@ -0,0 +1,143 @@
+//
+//  NoteListViewModel.swift
+//  CloudNotes
+//
+//  Created by 윤재웅 on 2021/06/03.
+//
+
+import CoreData
+import UIKit
+
+final class NoteManager: NSObject {
+    
+    static let shared = NoteManager()
+    var coreDataDelegate: CoreDataDelegate?
+    
+    // MARK: - Core Data stack
+    
+    lazy var persistentContainer: NSPersistentCloudKitContainer = {
+        
+        let container = NSPersistentCloudKitContainer(name: "CloudNotes")
+        container.loadPersistentStores(completionHandler: { (storeDescription, error) in
+            if let error = error as NSError? {
+                
+                fatalError("Unresolved error \(error), \(error.userInfo)")
+            }
+        })
+        return container
+    }()
+    
+    lazy var context: NSManagedObjectContext = {
+        return persistentContainer.viewContext
+    }()
+    
+    lazy var fetchedResultsController: NSFetchedResultsController<Note> = {
+        let fetchRequest: NSFetchRequest<Note> = Note.fetchRequest()
+        fetchRequest.sortDescriptors = [NSSortDescriptor(key: "lastModify", ascending: false)]
+        let fetchResult = NSFetchedResultsController(fetchRequest: fetchRequest,
+                                                     managedObjectContext: context,
+                                                     sectionNameKeyPath: nil,
+                                                     cacheName: nil)
+        fetchResult.delegate = self
+        return fetchResult
+    }()
+    
+    private override init() { }
+    
+    // MARK: - Core Data Saving support
+    
+    private func saveContext () {
+        let context = persistentContainer.viewContext
+        if context.hasChanges {
+            do {
+                try context.save()
+            } catch {
+                print(CoreDataError.save(error).errorDescription ?? "")
+            }
+        }
+    }
+    
+    private func save() {
+        do {
+            try context.save()
+        } catch {
+            print("Save Error")
+        }
+    }
+    
+    func specify(_ index: IndexPath?) -> Note {
+        guard let index = index else { return Note() }
+        if (fetchedResultsController.fetchedObjects?.count ?? 0) - 1 < index.row {
+            print(CoreDataError.indexPath.errorDescription ?? "")
+            return Note()
+        }
+        return fetchedResultsController.object(at: index)
+    }
+    
+    func count() -> Int {
+        return fetchedResultsController.fetchedObjects?.count ?? 0
+    }
```

Comment:
computed property를 사용할수도 있겠군요.
```suggestion
    var count: Int {
        return fetchedResultsController.fetchedObjects?.count ?? 0
    }
```